tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="DatabaseMonitoringPropagator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.DatabaseMonitoring
+{
+    internal static class DatabaseMonitoringPropagator
+    {
+        private const string SqlCommentSpanService = "dddbs";
+        private const string SqlCommentRootService = "ddps";
+        private const string SqlCommentVersion = "ddpv";
+        private const string SqlCommentEnv = "dde";
+
+        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, Span span)
+        {
+            if (propagationStyle == DbmPropagationLevel.Disabled)
+            {
+                return string.Empty;
+            }
+
+            var propagatorSringBuilder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
+            propagatorSringBuilder.Append($"{SqlCommentRootService}='{configuredServiceName}',{SqlCommentSpanService}='{span.ServiceName}'");
+
+            if (span.GetTag(Tags.Version) != null)
+            {
+                propagatorSringBuilder.Append($",{SqlCommentVersion}='{span.GetTag(Tags.Version)}'");
+            }
```

Comment:
nitpicking - You can also use this notation, it checks for nullity and avoids calling twice the method 
```suggestion
            if (span.GetTag(Tags.Version) is { } versionTag)
            {
                propagatorSringBuilder.Append($",{SqlCommentVersion}='{versionTag}'");
            }
```